src/Talis/Persona/Client/Tokens.php

```diff
@@ -57,34 +61,41 @@ public function validateToken($params = [])
      */
     protected function validateTokenUsingJWT($token, $scopes, $cacheTTL = 300)
     {
-        $cert = $this->retrieveJWTCertificate($cacheTTL);
+        $rawCert = $this->retrieveJWTCertificate($cacheTTL);
+
         try {
-            $decoded = (array)JWT::decode($token, $cert, ['RS256']);
+            // JWT::decode calls openssl_verify which will cause a fatal error
+            // if the certificate is invalid. Calling openssl_pkey_get_public
+            // first ensures that the certificate is valid before progressing.
+
+            if ($cert = openssl_pkey_get_public($rawCert)) {
+                $decoded = (array)JWT::decode($token, $cert, ['RS256']);
+            } else {
+                throw new \InvalidArgumentException('cannot parse public key');
```

Comment:
Are we throwing this here to catch it 5ish lines down? Could we not do what's done in that catch block here and not throw the exception?

Reply:
Less code, more precise.

Reply:
Ummm - Throwing exceptions for flow of control is usually a smell. But I'm not going to argue.